R/zchunk_L142.pfc_R_S_T_Y.R

```diff
@@ -8,83 +8,179 @@
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_SF6
+    get_data(all_data, "emissions/EDGAR/EDGAR_C2F6") %>%
+      gather(year, value, -IPCC_Annex, -World_Region, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_C2F6
+    get_data(all_data, "emissions/EDGAR/EDGAR_CF4") %>%
+      gather(year, value, -IPCC_Annex, -World_Region, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_CF4
+
+    # This chunk maps EDGAR HFC emissions to GCAM technologies
+    # First, create a table with all EDGAR HFCs, and prep "Other_F" table for use by renaming column.
+
+    EDGAR_SF6 %>%
+      mutate(Non.CO2 = "SF6") ->
+      EDGAR_SF6
+
+    EDGAR_C2F6 %>%
+      mutate(Non.CO2 = "C2F6") ->
+      EDGAR_C2F6
+
+    EDGAR_CF4 %>%
+      mutate(Non.CO2 = "CF4") %>%
+      bind_rows(EDGAR_SF6, EDGAR_C2F6,.) ->
+      ALL_EDGAR_HFC
+
+    Other_F %>%
+      rename(Non.CO2 = Gas) ->
+      Other_F
+
+    # Then, prepare EDGAR data for use.
+    # Map in region ID and sector name, and remove year 1970 to match "HISTORICAL_YEARS" constant.
+
+    ALL_EDGAR_HFC %>%
+      left_join_error_no_match(EDGAR_sector, by = "IPCC_description") %>%
+      standardize_iso(col = "ISO_A3") %>%
+      change_iso_code('rou','rom') %>%                                        # Switch Romania iso code to its pre-2002 value
+      left_join_error_no_match(iso_GCAM_regID, by = "iso")  %>%
+      rename(EDGAR_agg_sector = agg_sector) %>%
+      select(GCAM_region_ID, EDGAR_agg_sector, Non.CO2, year, value)  %>%
+      filter(year !="1970") ->
+      L142.EDGAR_HFC
+
+    L142.EDGAR_HFC %>%
+      group_by(GCAM_region_ID, EDGAR_agg_sector, Non.CO2, year) %>%
+      summarize(value = sum(value))  %>%
+      rename(EDGAR_emissions = value) ->
+      L142.EDGAR_PFC_R_S_T_Yh.tmp1
+
+    # Map in other f-gas sector, which varies by gas.
+    # Seperate PFCs and add non-CO2 gas information, then recombine.
+
+    L142.EDGAR_PFC_R_S_T_Yh.tmp1 %>%
+      filter(EDGAR_agg_sector != "other_f_gases") ->
+      L142.EDGAR_PFC_R_S_T_Yh_rest
+
+    L142.EDGAR_PFC_R_S_T_Yh.tmp1 %>%
+      filter(EDGAR_agg_sector == "other_f_gases")  %>%
+      ungroup() %>%
+      left_join(Other_F, by = "Non.CO2")   %>%
+      select(-EDGAR_agg_sector)  %>%
+      rename(EDGAR_agg_sector = Sector)  %>%
+      bind_rows(L142.EDGAR_PFC_R_S_T_Yh_rest,.)  %>%
+      ungroup() ->                                              # ungroup needed for later `left_join_keep_first_only`
+      L142.EDGAR_PFC_R_S_T_Yh.tmp1
+
+    # Map PFC gases to GCAM technologies.
+
+    GCAM_tech %>%
+      repeat_add_columns(tibble(GCAM_region_ID = GCAM_region_names$GCAM_region_ID)) %>%
+      repeat_add_columns(tibble(year = emissions.EDGAR_YEARS)) %>%
+      repeat_add_columns(tibble("Non.CO2" = emissions.PFCS))  %>%
```

Comment:
When I combine them I get an error message:
`Error: Variables must be length 1 or 38.
Problem variables: 'GCAM_region_ID', 'Non.CO2'`